src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -95651,6 +95648,29 @@ class Box<T>
                 Diagnostic(ErrorCode.ERR_BadBinaryOps, "x == null").WithArguments("==", "T", "<null>").WithLocation(53, 13));
         }
 
+        [Fact, WorkItem(34942, "https://github.com/dotnet/roslyn/issues/34942")]
+        public void ConditionalExpression_InferredNullability()
```

Comment:
Please add some more tests, especially around using these comparisons in more complicated expressions (where this is `&&`/`||` with something else) and in ternary operators.

Reply:
Haven't added a test using `||` (yet?) but we did learn some new things about ternary operators.

Reply:
Done